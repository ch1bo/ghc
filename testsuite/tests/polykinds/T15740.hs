{-# LANGUAGE RankNTypes, TypeInType, TypeFamilies #-}

module T15740 where

import Data.Kind

type family F2 :: forall k. k -> Type
data SBool :: Bool -> Type
data Nat

data SNat :: Nat -> Type

-- These should fail
type instance F2 = SBool
--type instance F2 = SNat
