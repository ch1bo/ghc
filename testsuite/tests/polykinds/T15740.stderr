
T15740.hs:14:20: error:
    • Expected kind ‘forall k. k -> *’,
        but ‘SBool’ has kind ‘Bool -> *’
    • In the type ‘SBool’
      In the type instance declaration for ‘F2’
